MiniApp/Classes/core/MiniAppInfoFetcher.swift

```diff
@@ -41,4 +41,39 @@ internal class MiniAppInfoFetcher {
             }
         }
     }
+
+    func getMiniAppMetaInfo(miniAppId: String, miniAppVersion: String, apiClient: MiniAppClient, completionHandler: @escaping (Result<MiniAppManifest, Error>) -> Void) {
+
+        apiClient.getMiniAppMetaData(appId: miniAppId, versionId: miniAppVersion) { (result) in
+            switch result {
+            case .success(let responseData):
+                if let decodeResponse = ResponseDecoder.decode(decodeType: MetaDataResponse.self,
+                                                               data: responseData.data) {
+                    return completionHandler(.success(self.prepareMiniAppManifest(
+                                                        metaDataResponse: decodeResponse)))
+                }
+                return completionHandler(.failure(NSError.invalidResponseData()))
+            case .failure(let error):
+                return completionHandler(.failure(error))
+            }
+        }
+    }
+
+    func prepareMiniAppManifest(metaDataResponse: MetaDataResponse) -> MiniAppManifest {
+        var manifest = MiniAppManifest()
+        manifest.requiredPermissions = metaDataResponse.reqPermissions?.compactMap {
+            guard let name = $0.name, let reason = $0.reason else {
```

Comment:
I think that `reason` should be optional. I know that it's required in the [W3C Spec](https://w3c.github.io/miniapp/specs/manifest/#dfn-reqpermissions), but I don't think we're actually going to use it. So I think it will cause a lot of confusion if we make it required.

I think we already made `permissionRequestDescription` optional, right?

Reply:
Done